scipy/spatial/distance.py

```diff
@@ -1374,7 +1376,13 @@ def pdist(X, metric='euclidean', p=None, w=None, V=None, VI=None):
         raise ValueError('A 2-dimensional array must be passed.')
 
     m, n = s
-    dm = np.zeros((m * (m - 1)) // 2, dtype=np.double)
+    if out is None:
+      dm = np.zeros((m * (m - 1)) // 2, dtype=np.double)
```

Comment:
It should be safe also to change this to  `dm = np.empty((m * (m - 1)) // 2, dtype=np.double)`